fullstack-cert/typescript-projects/fortune-app/src/script.ts

```diff
@@ -0,0 +1,265 @@
+/**
+ * Tarot Fortune App (TypeScript)
+ * - Fetches tarot card data and renders single or multi-card readings
+ * - Handles UI interactions and updates the fortune description panel
+ * - Keeps DOM access centralized and minimal for clarity and safety
+ */
+//import defaultImg from "bundle-text:./default.svg";
+const defaultImg = new URL("default.svg", import.meta.url);
+/** Shape of a tarot card returned by the remote dataset */
+interface Card {
+  name: string;
+  value: string | number;
+  name_short: string;
+  value_int: number;
+  suit: string;
+  type: string;
+  img: string;
+  meaning_up: string;
+  meaning_rev: string;
+  desc: string;
+}
+
+/** Deck is the root object returned by the API */
+interface Deck {
+  cards: Card[];
+}
+
+/** Base URL for remote images (PNG); used when cards provide an image path */
+const CDN_URL = "https://cdn.freecodecamp.org/curriculum/typescript/tarot-app";
+
+/** Utility to safely select DOM elements; throws if selector is missing */
+const getElement = <T extends HTMLElement>(selector: string): T => {
+  const el = document.querySelector(selector);
+  if (!el) throw new Error(`Element not found: ${selector}`);
+  return el as T;
```

Comment:
```suggestion
  const el = document.querySelector<T>(selector);
  if (!el) throw new Error(`Element not found: ${selector}`);
  return el;
```

Pass the generic instead of coercing.